covenant-signer/cmd/signerCmd.go

```diff
@@ -0,0 +1,73 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/babylonlabs-io/covenant-emulator/covenant-signer/config"
+	"github.com/babylonlabs-io/covenant-emulator/covenant-signer/keystore/cosmos"
+	m "github.com/babylonlabs-io/covenant-emulator/covenant-signer/observability/metrics"
+	"github.com/babylonlabs-io/covenant-emulator/covenant-signer/signerapp"
+	"github.com/babylonlabs-io/covenant-emulator/covenant-signer/signerservice"
+)
+
+func init() {
+	rootCmd.AddCommand(runSignerCmd)
+}
+
+var runSignerCmd = &cobra.Command{
+	Use:   "start",
+	Short: "starts the signer service",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		configPath, err := cmd.Flags().GetString(configPathKey)
+		if err != nil {
+			return err
+		}
+		cfg, err := config.GetConfig(configPath)
+		if err != nil {
+			return err
+		}
+
+		parsedConfig, err := cfg.Parse()
+
+		if err != nil {
+			return err
+		}
+
+		var prk signerapp.PrivKeyRetriever
+		if parsedConfig.KeyStoreConfig.KeyStoreType == config.CosmosKeyStore {
+			kr, err := cosmos.NewCosmosKeyringRetriever(parsedConfig.KeyStoreConfig.CosmosKeyStore)
+			if err != nil {
+				return err
+			}
+			prk = kr
+		} else {
+			return fmt.Errorf("unknown key store type")
+		}
+
+		app := signerapp.NewSignerApp(
+			prk,
+		)
+
+		metrics := m.NewCovenantSignerMetrics()
+
+		srv, err := signerservice.New(
+			cmd.Context(),
+			parsedConfig,
+			app,
+			metrics,
+		)
+
+		if err != nil {
+			return err
+		}
+
+		metricsAddress := fmt.Sprintf("%s:%d", cfg.Metrics.Host, cfg.Metrics.Port)
+
+		m.Start(metricsAddress, metrics.Registry)
+
+		// TODO: Add signal handling and gracefull shutdown
```

Comment:
Let's start using context instead of `chan` and signal interrupts (see vigilante or benchmark repo). Can be done in separate PR

Reply:
agree on that though I would to in separate pr 👍